### PR TITLE
gomobile: 2021-06-14 -> 2022-05-18

### PR DESCRIPTION
**Pull request by jakubgs**
###### Description of changes

This upgrade includes several important fixes for macOS and Catalyst, Android NDKs, and iOS builds.

- https://go.googlesource.com/mobile/+/8578da9835fd365e78a6e63048c103b27a53a82c
- https://go.googlesource.com/mobile/+/76c259c465ba39f84de7e2751a666612ddca556b
- https://go.googlesource.com/mobile/+/9cba7bc03c0fdba1b8fe7dfd476c2001656dde9e
- https://go.googlesource.com/mobile/+/80cb72e7af8e71b65e7d9e59fb4ec5d620134608
- https://go.googlesource.com/mobile/+/6d8ad35e460371e979b330d80492cba19676f933
- https://go.googlesource.com/mobile/+/b5eb1abba0f2d9331b29a29967d535890e026cec

Also moves use of `wrapProgram` to `postFixup` and dropped `ANDROID_NDK_HOME`.
No longer necessary as `8578da98` detects it from `ANDROID_HOME`.

This change has been tested through our iOS and Android builds in https://github.com/status-im/status-react/pull/13347.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/mobile/gomobile/default.nix [hunk 1]
```diff
@@ -43,19 +42,25 @@ buildGoModule {
   postInstall = ''
     mkdir -p $out/src/golang.org/x
     ln -s $src $out/src/golang.org/x/mobile
-    wrapProgram $out/bin/gomobile \
+  '';
+
+  fixupPhase = ''
```
Comment (SuperSandro2000) [inline pkgs/development/mobile/gomobile/default.nix:47]: We shouldn't overwrite the entire phase.

```suggestion
  postFixup = ''
```

File: pkgs/development/mobile/gomobile/default.nix [hunk 2]
```diff
@@ -43,19 +42,25 @@ buildGoModule {
...
-      --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath [ zlib ]}"
+        --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath [ zlib ]}"
+    done
   '';
 
   meta = with lib; {
     description = "A tool for building and running mobile apps written in Go";
     homepage = "https://pkg.go.dev/golang.org/x/mobile/cmd/gomobile";
     license = licenses.bsd3;
+    platforms = platforms.unix;
```
Comment (SuperSandro2000) [inline pkgs/development/mobile/gomobile/default.nix:63]:
```suggestion
```

that is the default of buildGoModule

Comment (jakubgs) [reply pkgs/development/mobile/gomobile/default.nix]: Ah, indeed, thanks.

File: pkgs/development/mobile/gomobile/default.nix (hunk 1 above)
Comment (jakubgs) [reply pkgs/development/mobile/gomobile/default.nix]: Good point.

**Pull request merged**